### PR TITLE
[Fix #1637] Integrate with savehist-mode

### LINKED ISSUE #1637: Commit 0d52268 has broken storing the compile history between Emacs runs
The compile history is no longer stored between Emacs runs.
The problem was introduced with commit 0d52268, issue #1601.

## Expected behavior

- `projectile-compile-project` command history should be saved on disk (by default in `~/emacs.d/history`)
- When Emacs is closed and started again, `projectile-compile-project` should load the history back from disk.

## Actual behavior

- The history from `projectile-compile-project` isn't saved.
- Even if there's previous history from before this problem started occurring, this history won't be loaded.
- The plain `compile` command loads/saves history just fine.

## Steps to reproduce the problem

1. Open any .cpp file from a C++ project in Emacs (my file path, for reference: `/mnt/n/dev/4walls/4walls/src/core/ecs/flags.cpp`)
2. `projectile-compile-project`
3. Customize the compile command.
4. Execute it.
5. Close Emacs.
6. Run Emacs again, reopen the same .cpp file.
7. `projectile-compile-project`, try to go back in history.

## Environment & Version information

I use Helm `20210111.1454`.

### Projectile version information

```
Projectile 20210104.1216
```

### Emacs version

```
GNU Emacs 27.1 (build 1, x86_64-pc-linux-gnu, GTK+ Version 3.24.20)
 of 2020-10-07
```

### Operating system

Kubuntu 20.04

### PR DESCRIPTION
**Pull request by TxGVNN**
Hi @bbatsov 

I would like to add a PR to fix #1637. Could you have a look?

Thanks

-----------------

Before submitting a PR make sure the following things have been done (and denote this
by checking the relevant checkboxes):

- [x] The commits are consistent with our [contribution guidelines](../blob/master/CONTRIBUTING.md)
- [x] You've added tests (if possible) to cover your change(s)
- [ ] All tests are passing ([`eldev test`](https://github.com/doublep/eldev))
- [x] The new code is not generating bytecode or `M-x checkdoc` warnings
- [x] You've updated the [changelog](../blob/master/CHANGELOG.md) (if adding/changing user-visible functionality)
- [ ] You've updated the readme (if adding/changing user-visible functionality)

Thanks!

### CODE REVIEW

File: projectile.el [hunk 1]
```diff
@@ -5125,6 +5125,13 @@ Otherwise behave as if called interactively.
     (advice-remove 'compilation-find-file #'compilation-find-file-projectile-find-compilation-buffer)
     (advice-remove 'delete-file #'delete-file-projectile-remove-from-cache))))
 
+(if (bound-and-true-p savehist-loaded)
```
Comment (bbatsov) [inline projectile.el:5128]: I think it'd be best to move such setup to `projectile-mode`.

Comment (TxGVNN) [reply projectile.el]: Yes sure. I updated

File: projectile.el [hunk 2]
```diff
@@ -5118,7 +5118,13 @@ Otherwise behave as if called interactively.
     (add-hook 'projectile-find-dir-hook #'projectile-track-known-projects-find-file-hook t)
     (add-hook 'dired-before-readin-hook #'projectile-track-known-projects-find-file-hook t t)
     (advice-add 'compilation-find-file :around #'compilation-find-file-projectile-find-compilation-buffer)
-    (advice-add 'delete-file :before #'delete-file-projectile-remove-from-cache))
+    (advice-add 'delete-file :before #'delete-file-projectile-remove-from-cache)
+    (if (bound-and-true-p savehist-loaded)
```
Comment (bbatsov) [inline projectile.el:5122]: Oh second, thought - this is not ideal as we'll keep doing the same thing every time the mode starts. Probably it's better to have this at the top-level as initially proposed, just add some comments there explaining the need for this, as it's unlikely that even I'll remember this several months down the road.

Comment (TxGVNN) [reply projectile.el:5122]: As I see, even we keep doing the same thing when mode starts but seems it doesn't affect anything. Anyway, up to you. Do you want to revert this commit?

Comment (TxGVNN) [reply projectile.el:5122]: HI @bbatsov, any chance?

Comment (bbatsov) [reply projectile.el:5122]: Yeah, I think it's better to revert it and document better the previous solution.

Comment (TxGVNN) [reply projectile.el]: I updated. Hope it is good.

**Comment by bbatsov** [comment]
Thanks!

**Pull request merged**